### PR TITLE
Disable unnecessary OpenID/OAuth2 elements

### PR DESCRIPTION
**Pull request by pboguslawski**
This mod fixes disabling unnecessary OpenID/OAuth2 elements.

Related: https://github.com/go-gitea/gitea/pull/13129
Author-Change-Id: IB#1115256

### CODE REVIEW

File: routers/web/web.go [hunk 1]
```diff
@@ -277,9 +277,15 @@ func RegisterRoutes(m *web.Route) {
 			m.Get("/{provider}", auth.SignInOAuth)
 			m.Get("/{provider}/callback", auth.SignInOAuthCallback)
 		})
-		m.Get("/link_account", auth.LinkAccount)
-		m.Post("/link_account_signin", bindIgnErr(forms.SignInForm{}), auth.LinkAccountPostSignIn)
-		m.Post("/link_account_signup", bindIgnErr(forms.RegisterForm{}), auth.LinkAccountPostRegister)
+		m.Group("/link_account", func() {
+			m.Get("", auth.LinkAccount)
+		}, openIDSignInEnabled)
```
Comment (Gusted) [inline routers/web/web.go]: `openIDSignInEnabled` can be added to the `Get` and `Post` arguments, no need to go all fancy with `Group` here.

Comment (pboguslawski) [reply routers/web/web.go]: Just tested that
```
m.Get("/", Home, openIDSignInEnabled)
```
does not forbid access to / when openIDSignInEnabled is disabled but

```
m.Group("/", func() {
		m.Get("", Home)
	}, openIDSignInEnabled)
```
does. Seems that simply adding `openIDSignInEnabled` to `Get` arguments does not work.

Comment (pboguslawski) [reply routers/web/web.go]:
```
m.Get("/", openIDSignInEnabled, Home)
```
works ok - will fix it.

Comment (pboguslawski) [reply routers/web/web.go]: Done.

**Review by Gusted — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Review by lunny — CHANGES_REQUESTED** [review]
LinkAccount is not only for openID, OAuth2 could also use that. This is totally wrong.

**Comment by pboguslawski** [comment]
> LinkAccount is not only for openID, OAuth2 could also use that. This is totally wrong.

Fixed in 513303bd6d3be8eb8760d2b352a48423650858f3.

**Review by lunny — APPROVED** [review]

**Review by Gusted — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by wxiaoguang** [comment]
@pboguslawski this change (maybe) causes regressions like:

* Oauth2 Provider must be enabled for Account Linking by login to work on Oauth2 Client #35744
*  ~~ACCOUNT_LINKING depends on ENABLE_AUTO_REGISTRATION #27709~~ (not quite sure about this one)

Could you propose a proper fix?

(I will propose a quick fix to revert this PR for a temp approach)

**Pull request merged**